server/routes/viewingLicences/handlers/printLicence.ts

```diff
@@ -0,0 +1,81 @@
+import { Request, Response } from 'express'
+import logger from '../../../../logger'
+import config from '../../../config'
+import PrisonerService from '../../../services/prisonerService'
+
+const pdfHeaderFooterStyle =
+  'font-family: Arial; ' +
+  'font-size: 10px; ' +
+  'font-weight: normal; ' +
+  'width: 100%; ' +
+  'height: 35px; ' +
+  'text-align: center; ' +
+  'padding: 20px;'
+
+export default class PrintLicenceRoutes {
+  constructor(private readonly prisonerService: PrisonerService) {}
+
+  preview = async (req: Request, res: Response): Promise<void> => {
+    const { username } = res.locals.user
+    const { licence } = res.locals
+    const htmlPrint = true
+    logger.info(`HTML preview licence ID [${licence.id}] type [${licence.typeCode}] by user [${username}]`)
+    res.render(`pages/licence/${licence.typeCode}`, { licence, htmlPrint })
+  }
+
+  renderPdf = async (req: Request, res: Response): Promise<void> => {
+    const { username } = res.locals.user
+    const { licence } = res.locals
+
+    // Specify licencesUrl for assets so that it is used in the NJK template as http://host.docker.internal:3000/assets
+    const { licencesUrl, pdfOptions } = config.apis.gotenberg
+
+    const imageData = await this.prisonerService.getPrisonerImageData(username, licence.nomsId)
+    const filename = licence.nomsId ? `${licence.nomsId}.pdf` : `${licence.lastName}.pdf`
+    const footerHtml = this.getPdfFooter(
+      licence.nomsId,
+      licence.cro,
+      licence.bookingNo,
+      licence.pnc,
+      licence.typeCode,
+      licence.id,
+      licence.version,
+      licence.prisonCode
+    )
+
+    logger.info(`PDF print licence ID [${licence.id}] type [${licence.typeCode}] by user [${username}]`)
+
+    res.renderPDF(
+      `pages/licence/${licence.typeCode}`,
+      { licencesUrl, licence, imageData, htmlPrint: false },
+      { filename, pdfOptions: { headerHtml: null, footerHtml, ...pdfOptions } }
+    )
+  }
+
+  getPdfFooter = (
+    nomsId: string,
+    cro: string,
+    bookingNo: string,
+    pnc: string,
+    licenceType: string,
+    licenceId: string,
+    version: string,
+    prison: string
+  ): string => {
+    return `
```

Comment:
No way to make this a nunjucks partial?

Reply:
Unfortunately not, its sent as a separate HTML document directly into Gotenberg. However, I do want to experiment with a CSS-only footer as a separate task.